metricbeat/module/windows/perfmon/reader.go

```diff
@@ -124,16 +124,14 @@ func (r *Reader) RefreshCounterPaths() error {
 	if err != nil {
 		return errors.Wrap(err, "failed removing unused counter values")
 	}
-	// Some counters, such as rate counters, require two counter values in order to compute a displayable value. In this case we must call PdhCollectQueryData twice before calling PdhGetFormattedCounterValue.
-	// For more information, see Collecting Performance Data (https://docs.microsoft.com/en-us/windows/desktop/PerfCtrs/collecting-performance-data).
-	if err = r.query.CollectData(); err != nil {
-		return errors.Wrap(err, "failed querying counter values")
-	}
+
 	return nil
 }
 
 // Read executes a query and returns those values in an event.
 func (r *Reader) Read() ([]mb.Event, error) {
+	// Some counters, such as rate counters, require two counter values in order to compute a displayable value. In this case we must call PdhCollectQueryData twice before calling PdhGetFormattedCounterValue.
```

Comment:
Where are we calling `CollectData` a second time now?

Reply:
This will happen the second time will execute the Read method.  For any new resulting counters from the Refresh method this will happen again the second time the Read method is called.
Let me know if there are any better options to delay the second CollectData call between the Refresh and Read methods.

Reply:
Ok, for the comment I thought we needed to explicitly call `CollectData` twice for each result. If  it is enough with the call on each `Read` then it looks good to me. Actually the code looks better this way.